vm/actor/tests/puppet_actor_test.rs

```diff
@@ -72,5 +72,18 @@ fn simple_send() {
 
     assert!(ret.is_ok());
 
-    assert_eq!(exp_ret, ret.unwrap().return_bytes);
+    assert_eq!(Some(exp_ret), ret.unwrap().return_bytes);
+}
+
+#[test]
+fn serialize_test() {
+    let mut v: Vec<FailToMarshalCBOR> = vec![];
+
+    // Should pass becuase vec is empty
+    assert!(Serialized::serialize(&v).is_ok());
+
+    v.push(FailToMarshalCBOR::default());
+
+    // Should fail becuase vec is no longer empty
+    assert!(Serialized::serialize(v).is_err());
```

Comment:
Can you also check the case for `runtime_transaction_marshal_cbor_failure`? my intuition is that it won't error as it's expected to